synapse/rest/admin/rooms.py

```diff
@@ -150,21 +154,21 @@ async def on_GET(
                 HTTPStatus.BAD_REQUEST, "%s is not a legal room ID" % (room_id,)
             )
 
-        delete_ids = self._pagination_handler.get_delete_ids_by_room(room_id)
-        if delete_ids is None:
-            raise NotFoundError("No delete task for room_id '%s' found" % room_id)
+        delete_statuses = await self._pagination_handler.get_delete_statuses_by_room(
+            room_id
+        )
 
         response = []
-        for delete_id in delete_ids:
-            delete = self._pagination_handler.get_delete_status(delete_id)
-            if delete:
-                response += [
-                    {
-                        "delete_id": delete_id,
-                        **delete.asdict(),
-                    }
-                ]
-        return HTTPStatus.OK, {"results": cast(JsonDict, response)}
+        for delete_status in delete_statuses:
+            # We ignore scheduled deletes because currently they are only used
+            # for automatically purging forgotten room after X time.
```

Comment:
```suggestion
            # We ignore scheduled deletes because currently they are only used
            # for automatically purging forgotten room after X time and this API
            # is only concerned with exposing manual actions from the admins
```